### DISCUSSION TITLE
Add css dynamically

### DISCUSSION THREAD

**Discussion by jacopsd** [GrapesJS/grapesjs#5276] [Q&A]
How can css code be added dynamically to the canvas?

I know the editor init function has the canvas 'styles' option, but that only accepts css files, not css code itself:

`
let editor = grapesjs.init({
            canvas: {
                styles: [
                   '/static/dir1/my_style.css',
            ]
       }
`

I tried [css.setRule()](https://grapesjs.com/docs/api/css_composer.html#setrule) but for some reason the css is not applied to my loaded page inside grapesjs

**Comment by artf** [comment]
You should be able to add dynamic CSS via [css.addRules](https://grapesjs.com/docs/api/css_composer.html#addrules). 
In case something doesn't work as expected, please provide a reproducible demo.
